core/trino-main/src/main/java/io/trino/metadata/MetadataUtil.java

```diff
@@ -111,30 +113,34 @@ public static String getSessionCatalog(Metadata metadata, Session session, Node
         return catalog;
     }
 
-    public static CatalogSchemaName createCatalogSchemaName(Session session, Node node, Optional<QualifiedName> schema)
+    public static CatalogSchemaName createCatalogSchemaName(Session session, Node node, Optional<QualifiedName> schema, Metadata metadata)
     {
-        String catalogName = session.getCatalog().orElse(null);
-        String schemaName = session.getSchema().orElse(null);
```

Comment:
can we assume that schema and catalog in session are in canonical form?

Reply:
Yes !! That is the assumption

Reply:
So there is no need to change the existing code.